cloud/pod_util.go

```diff
@@ -262,22 +258,21 @@ func exportECSPodExecutionOptions(ecsConfig evergreen.ECSConfig, containerOpts p
 			SetSecurityGroups(ecsConfig.AWSVPC.SecurityGroups))
 	}
 
-	placementOpts := cocoa.NewECSPodPlacementOptions()
-	if containerOpts.WindowsVersion != "" {
-		windowsVersionConstraint := fmt.Sprintf("%s == %s", ecsWindowsVersionTagConstraint, containerOpts.WindowsVersion)
-		placementOpts.AddInstanceFilters(windowsVersionConstraint)
-	}
-	opts.SetPlacementOptions(*placementOpts)
-
 	var foundCapacityProvider bool
 	for _, cp := range ecsConfig.CapacityProviders {
+		if containerOpts.OS == pod.OSWindows && !containerOpts.WindowsVersion.Matches(cp.WindowsVersion) {
+			continue
+		}
 		if containerOpts.OS.Matches(cp.OS) && containerOpts.Arch.Matches(cp.Arch) {
 			opts.SetCapacityProvider(cp.Name)
 			foundCapacityProvider = true
 			break
 		}
 	}
 	if !foundCapacityProvider {
+		if containerOpts.OS == pod.OSWindows {
+			return nil, errors.Errorf("container OS '%s' with Windows version '%s' and arch '%s' did not match any recognized capacity provider", containerOpts.OS, containerOpts.WindowsVersion, containerOpts.Arch)
```

Comment:
Since we know OS is `OSWindows` here it may be redundant to say "with Windows version"